nuxeo-ai-core/src/main/java/org/nuxeo/ai/configuration/ThresholdConfiguratorDescriptor.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * (C) Copyright 2006-2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * Contributors:
+ *     anechaev
+ */
+package org.nuxeo.ai.configuration;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import org.nuxeo.common.xmap.annotation.XNode;
+import org.nuxeo.common.xmap.annotation.XNodeList;
+import org.nuxeo.common.xmap.annotation.XObject;
+import org.nuxeo.runtime.model.Descriptor;
+
+@XObject("thresholdConfiguration")
+public class ThresholdConfiguratorDescriptor implements Descriptor {
+
+    @XNode("@global")
+    protected float global;
+
+    @XNodeList(value = "configurations/configuration", type = ArrayList.class, componentType = Configuration.class)
+    protected List<Configuration> configurations = new ArrayList<>();
+
+    public float getGlobal() {
+        return global;
+    }
+
+    public List<Configuration> getConfigurations() {
+        return configurations;
+    }
+
+    @Override
+    public String getId() {
+        return UUID.randomUUID().toString();
+    }
+
+    @XObject("configuration")
+    public static class Configuration {
+
+        @XNode("@type")
+        protected String type;
+
+        @XNodeList(value = "thresholds/threshold", type = ArrayList.class, componentType = Threshold.class)
+        protected List<Threshold> thresholds = new ArrayList<>();
+
+        /**
+         * @return String as a Document Type or Facet
+         */
+        public String getType() {
+            return type;
+        }
+
+        /**
+         * @return List of {@link Threshold} objects
+         */
+        public List<Threshold> getThresholds() {
+            return thresholds;
+        }
+    }
+
+    @XObject("threshold")
+    public static class Threshold {
+
+        @XNode("@xpath")
+        protected String xpath;
+
+        @XNode("@value")
+        protected float value = 0.f;
+
+
+        @XNode("@autofill")
+        protected float autofillValue = 0.f;
+
+        @XNode("@autocorrect")
+        protected float autocorrect = 0.f;
+
+        public void converge(Threshold that) {
```

Comment:
is this `merge`?  perhaps some javadoc?

Reply:
it's like a merge but with evolving to maximum. if converge is not the right word then ok.
javadoc - sure, my bad missed it